src/Factory/LoggerFactory.php

```diff
@@ -30,10 +31,14 @@ class LoggerFactory
 	 */
 	public static function create($channel, Configuration $config)
 	{
+		if(empty($config->get('system', 'debugging', false))) {
+			return new VoidLogger();
```

Comment:
> Where can't we return `null`?


If I replace this with `return null;`

```console
[Wed Feb 27 20:24:52.075285 2019] [:error] [pid 9905] [client 192.168.22.1:60600] PHP Fatal error:  Uncaught TypeError: Argument 2 passed to Friendica\\App::__construct() must be an instance of Psr\\Log\\LoggerInterface, null given, called in /vagrant/src/Factory/DependencyFactory.php on line 35 and defined in /vagrant/src/App.php:215\nStack trace:\n#0 /vagrant/src/Factory/DependencyFactory.php(35): Friendica\\App->__construct(Object(Friendica\\Core\\Config\\Configuration), NULL, Object(Friendica\\Util\\Profiler), false)\n#1 /vagrant/index.php(15): Friendica\\Factory\\DependencyFactory::setUp('index', '/vagrant', false)\n#2 {main}\n  thrown in /vagrant/src/App.php on line 215
```

And type-hint doesn't support `LoggerInterface|null` as parameter

Reply:
> And type-hint doesn't support `LoggerInterface|null` as parameter

It does if you define the parameter as optional with a `null` default value. Otherwise, you have to use the `LoggerInterface?` type-hint introduced in PHP 7, which would require us to bump the minimum technical version of Friendica.

Reply:
I'm for adding null to the parameter, because that shouldn't force us to bump the version.

Reply:
On the other hand, then you would have to check the value of `App->logger` before invoking any of its expected methods. The `VoidLogger` actually sounds cleaner to me.